core-api/src/main/java/com/sequenceiq/cloudbreak/api/endpoint/v4/stacks/response/recovery/RecoveryValidationV4Response.java

```diff
@@ -30,6 +35,25 @@ public void setStatus(RecoveryStatus status) {
         this.status = status;
     }
 
+    public RecoveryValidationV4Response merge(RecoveryValidationV4Response other) {
+        boolean nonRecoverable = status.nonRecoverable();
+        boolean otherNonRecoverable = other.getStatus().nonRecoverable();
+        RecoveryStatus mergedState = nonRecoverable || otherNonRecoverable ? NON_RECOVERABLE : RECOVERABLE;
+        String delimiter = mergedState.recoverable() ? "" : " Next issue: ";
```

Comment:
```java
        System.out.println(new RecoveryValidationV4Response("Success.", RecoveryStatus.RECOVERABLE)
                .merge(new RecoveryValidationV4Response("Success.", RecoveryStatus.RECOVERABLE)).getReason());
```

This results in the message `Success.Success.`. Can you please add a white space as the delimiter and maybe later trim the result?